File: server/models/invoice_model.js
const db = require("../db/db");
const invoiceSchema = new db.mongoose.Schema(
    {
        user_id: {
            type: String,
            required: true
        },
        total: {
            type: Number,
            require: true
        },
        date: {
            type: String,
            require: true
        },
        status: {
            type: String,
            require: true
        },
        payment_method: {
            type: String,
            require: true
        },
        delivery_address: {
            type: String,
            require: true
        },
        shipping_fee: {
            type: String,
            require: true
        },
        carrier_id: {
            type: db.mongoose.Schema.Types.ObjectId,
            required: true,
            ref: "carrier"
        },
        order_id: {
            type: db.mongoose.Schema.Types.ObjectId,
            required: false,
            ref: "order"
        },
        code: {
            type: String,
            required: false
        },
        create_at: {
            type: String,
            require: true
        }
    },
    {
        collection: "invoice",
        timestamps: true,
        versionKey: false
    }
);
const invoiceModel = db.mongoose.model("invoice", invoiceSchema);
module.exports = { invoiceModel };

File: app/src/main/java/com/example/petify/data/server/enitities/InvoiceModel.kt
package com.example.petify.data.server.enitities

import com.google.gson.annotations.SerializedName

data class InvoiceModel(
    @field:SerializedName("_id") var id: String = "",
    @field:SerializedName("user_id") var userId: String = "",
    @field:SerializedName("total") var total: Double = 0.0,
    @field:SerializedName("date") var date: String = "",
    @field:SerializedName("status") var status: String = "",
    @field:SerializedName("payment_method") var paymentMethod: String = "",
    @field:SerializedName("delivery_address") var deliveryAddress: String = "",
    @field:SerializedName("shipping_fee") var shippingFee: Double = 0.0,
    @field:SerializedName("supplier_id") var supplierId: String = "",
    @field:SerializedName("carrier_id") var carrierId: String = "",
    @field:SerializedName("order_id") var orderId: String = "",
    @field:SerializedName("create_at") var createAt : String = ""
)